store/store.go

```diff
@@ -2089,29 +2092,42 @@ func (s *Store) SnapAction(ctx context.Context, currentSnaps []*CurrentSnap, act
 	}
 }
 
+func genInstanceKey(curSnap *CurrentSnap, seed string) string {
+	_, snapInstanceKey := snap.SplitInstanceName(curSnap.InstanceName)
+
+	if snapInstanceKey == "" {
+		return curSnap.SnapID
+	}
+
+	// due to privacy concerns, avoid sending the local names to the
+	// backend, instead hash the snap ID and instance key together
+	h := crypto.SHA256.New()
+	// TODO parallel-install: include seed
+	h.Write([]byte(curSnap.SnapID))
+	h.Write([]byte(snapInstanceKey))
+	h.Write([]byte(seed))
+	enc := base64.RawURLEncoding.EncodeToString(h.Sum(nil))
+	return fmt.Sprintf("%s-%s", curSnap.SnapID, enc)
```

Comment:
as I said this need to be stable but I don't expect them to be parsed,  if the '-' separator vs in base64 encoding is confusing for debugging we could use a different separator  ':' or '.'